pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -395,3 +404,82 @@ func (ca *cloudCIDRAllocator) ReleaseCIDR(node *v1.Node) error {
 		node.Name, node.Spec.PodCIDR)
 	return nil
 }
+
+func (ca *cloudCIDRAllocator) updateMultiNetworkAnnotations(node *v1.Node, northInterfaces networkv1.NorthInterfacesAnnotation, additionalNodeNetworks networkv1.MultiNetworkAnnotation) error {
+	northInterfaceAnn, err := networkv1.MarshalNorthInterfacesAnnotation(northInterfaces)
+	if err != nil {
+		klog.ErrorS(err, "Failed to marshal the north interfaces annotation for multi-networking", "nodeName", node.Name)
+		return err
+	}
+	additionalNodeNwAnn, err := networkv1.MarshalAnnotation(additionalNodeNetworks)
+	if err != nil {
+		klog.ErrorS(err, "Failed to marshal the additional node networks annotation for multi-networking", "nodeName", node.Name)
+		return err
+	}
+	if node.Annotations == nil {
+		node.Annotations = make(map[string]string)
+	}
+	node.Annotations[networkv1.NorthInterfacesAnnotationKey] = northInterfaceAnn
+	node.Annotations[networkv1.MultiNetworkAnnotationKey] = additionalNodeNwAnn
+	node.Status.Capacity, err = allocateIPCapacity(node, additionalNodeNetworks)
+	if err != nil {
+		return err
+	}
+	// Prepare patch bytes for the node update.
+	patchBytes, err := json.Marshal([]interface{}{
+		map[string]interface{}{
+			"op":    "replace",
```

Comment:
merge or replace, does this wipe out all the annotations?

Reply:
We are replacing only the annotations we are interested in - `node.Annotations[networkv1.NorthInterfacesAnnotationKey]` and `node.Annotations[networkv1.MultiNetworkAnnotationKey]`.  So other annotations are not replaced. 

I've explained in the comment below (lines 441-442) why we are doing a JSON merge patch instead of a strategic merge patch. A JSON merge patch does not support a `merge` operation. (https://jsonpatch.com/#operations).

But there is a chance that we overwrite the annotations on the node that were written **after** we fetched the node here. I'm not sure how we can tackle that. I'm looking for suggestions here.

Reply:
that should generate a new event -> reconcile -> update with desired state